stored_requests/backends/db_provider/postgres_dbprovider.go

```diff
@@ -78,8 +82,41 @@ func (provider *PostgresDbProvider) ConnString() string {
 		buffer.WriteString(" ")
 	}
 
-	buffer.WriteString("sslmode=disable")
-	return buffer.String()
+	// TLS connection
+	var sslmode = "disable"
+
+	if provider.cfg.TLS.RootCert != "" {
+		sslmode = "verify-ca"
+
+		buffer.WriteString("sslrootcert=")
+		buffer.WriteString(provider.cfg.TLS.RootCert)
+		buffer.WriteString(" ")
+
+		if provider.cfg.TLS.ClientCert != "" && provider.cfg.TLS.ClientKey != "" {
+			sslmode = "verify-full"
+
+			buffer.WriteString("sslcert=")
+			buffer.WriteString(provider.cfg.TLS.ClientCert)
+			buffer.WriteString(" ")
+
+			buffer.WriteString("sslkey=")
+			buffer.WriteString(provider.cfg.TLS.ClientKey)
+			buffer.WriteString(" ")
+		}
+	}
+
+	if !strings.Contains(provider.cfg.QueryString, "sslmode=") {
+		buffer.WriteString("sslmode=")
+		buffer.WriteString(sslmode)
+	}
+
+	// Rest of connection string parameters passed through query_string
+	if provider.cfg.QueryString != "" {
+		buffer.WriteString(" ")
+		buffer.WriteString(provider.cfg.QueryString)
+	}
```

Comment:
It appears postgres supports two connection string formats: key/val strings and URIs ([34.1.1 Connection Strings](https://www.postgresql.org/docs/current/libpq-connect.html#LIBPQ-CONNSTRING)) We're using the key/val format in this function but we are then appending the query string to the end of it. I guess technically a host could specify the query string as key/val with spaces separating the pairs but given that the name of the field is `query_string`, I would expect them to be defined in a URI format with `&` separating the pairs.
I think we should do one of the following:
1. move to a URI format (I can't imagine there are any drawbacks to doing so but we should do more research first. I think this is my preferred path forward as it should be simplest)
2. gracefully handle a URI-formatted `query_string` parameter by converting `&` to a space
3. validate the query string parameter and provide some indication of how it should be formatted

I guess we could always just let the connection fail on startup and the host could figure it out but that's not great. We are doing that to some extent already though I believe.

Thoughts?

Reply:
Moved to a URI format as per the preferred option of the suggested ones.